app/presenters/concerns/hyrax/iiif_av/displays_content_decorator.rb

```diff
@@ -0,0 +1,76 @@
+module Hyrax
+  module IiifAv
+    module DisplaysContentDecorator
+      # main reasons for this decorator is to override variable names from hyrax-iiif_av
+      #   solr_document => object
+      #   current_ability => @ability
+      #   request.base_url => hostname
+      # also to remove #auth_service since it was not working for now
+
+      private
+
+        def solr_document
+          defined?(super) ? super : object
+        end
+
+        def current_ability
+          defined?(super) ? super : @ability
+        end
+
+        def request
+          OpenStruct.new(base_url: hostname)
+        end
```

Comment:
We really want to avoid OpenStruct, it's a vector for all kinds of memory issues.  In this case, it's minor.

Instead we should define a struct.

```ruby
Request = Struct.new(:base_url, keyword_init: true)

def request
  Request.new(base_url: hostname)
end
```

My question, and an explanatory comment would clarify.  Why is this hostname instead of base_url or vice versa.

Reply:
Yup that's the funky stuff i was referring to.  I was trying to get past all these `request.base_url` calls in the gem

https://github.com/samvera-labs/hyrax-iiif_av/blob/888ad531ec08e6c1040d609321972e2147f0c702/app/presenters/concerns/hyrax/iiif_av/displays_content.rb#L92